pkg/plugins/client_test.go

```diff
@@ -132,3 +134,83 @@ func TestClientScheme(t *testing.T) {
 		}
 	}
 }
+
+func TestNewClientWithTimeout(t *testing.T) {
+	addr := setupRemotePluginServer()
+	defer teardownRemotePluginServer()
+
+	m := Manifest{[]string{"VolumeDriver", "NetworkDriver"}}
+
+	mux.HandleFunc("/Test.Echo", func(w http.ResponseWriter, r *http.Request) {
+		time.Sleep(time.Second)
+	})
+
+	c, _ := NewClientWithTimeout(addr, &tlsconfig.Options{InsecureSkipVerify: true}, 1)
+	var output Manifest
+	err := c.Call("Test.Echo", m, &output)
+	if err == nil {
+		t.Fatal("Expected timeout error")
```

Comment:
If the handler sleeps for 1 second, and the timeout is exactly 1 second, isn't that potentially unreliable? I'd expect the handler to sleep slightly longer to make sure the timeout is reached.

I'd suggest changing `NewClientWithTimeout` to take a `time.Duration` instead of a number of seconds, so we can avoid sleeping for a full second. `time.Duration` is better stylistically, anyway.